File: ts/sync-api/sync_api_synchronize_changes.ts
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
import { createDeliveryClient } from "@kontent-ai/delivery-sdk";

const deliveryClient = createDeliveryClient({
  environmentId: "<YOUR_ENVIRONMENT_ID>",
});

const response = await deliveryClient.syncChanges().toPromise();

// Store the X-Continuation token in a persistent storage for later use
const continuationToken = response.xContinuationToken;

for (const item of response.data.items) {
  if (item.changeType === "changed") {
    // Process a changed item
  } else if (item.changeType === "deleted") {
    // Process a deleted item
  }
}

// Synchronize the next batch of changes
const nextResponse = await deliveryClient.syncChanges().withContinuationToken(continuationToken).toPromise();

File: net/sync-api/sync_api_synchronize_changes.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Delivery;

// Tip: Use DI to create Delivery client https://kontent.ai/learn/net-register-client
IDeliveryClient client = DeliveryClientBuilder
    .WithEnvironmentId("KONTENT_AI_ENVIRONMENT_ID")
    .Build();

// Gets a list of recently changed content items
IDeliverySyncResponse response = await client.GetSyncAsync("KONTENT_AI_CONTINUATION_TOKEN");
IList<ISyncItem> syncItems = response.SyncItems;
await client.GetSyncAsync("token");